test/uk/gov/hmrc/personaldetailsvalidation/audit/AuditDataEventFactorySpec.scala

```diff
@@ -45,7 +46,7 @@ class AuditDataEventFactorySpec extends UnitSpec with MockFactory {
         dataEvent.auditSource shouldBe auditConfig.appName
         dataEvent.auditType shouldBe "MatchingResult"
         dataEvent.tags shouldBe auditTags
-        dataEvent.detail shouldBe auditDetails + ("nino" -> personalDetails.nino.value) ++ matchingingDetails
+        dataEvent.detail shouldBe auditDetails + ("nino" -> personalDetails.nino.value) + ("postCode" -> "NOT SUPPLIED") ++ matchingingDetails
```

Comment:
I think we need to add two tests for the cases when the postcode is provided 
1.  successful matching 
2. failed matching